app/components/reports/region_tree_component.rb

```diff
@@ -0,0 +1,20 @@
+class Reports::RegionTreeComponent < ViewComponent::Base
+  attr_reader :parent, :children
+
+  def initialize(parent:, children:)
+    @children = children
+    @parent = parent
+  end
+
+  OFFSET = 2
+
+  def depth(region)
+    depth = Region::REGION_TYPES.index(region.region_type) - OFFSET
+    depth += 1 if region.child_region_type.nil?
```

Comment:
Can you tell me what this +1 is for?

Reply:
This is one reason we want this as a method and not as an arg passed in I think.

We want the depth to be based off the highest level region type, minus the offset, and then we need to add one for the case where we _don't_ display the dropdown icon ...which is the case where there aren't any child regions to show.

Reply:
Ahhh I see, is that because the dropdown icon bumps the text over to the right? That sounds like a view concern that we can solve with appropriate styling, since conceptually, the depth of one item shouldn't change based on whether it has children or not.

In any case, we're not going to do depth-in-method-call as you mentioned above ☝️ , so I'll let you decide if you've got the motivation to figure out the CSS for that 😅  No big deal either ways 👍

Reply:
> Ahhh I see, is that because the dropdown icon bumps the text over to the right? That sounds like a view concern that we can solve with appropriate styling, since conceptually, the depth of one item shouldn't change based on whether it has children or not.
> 
> In any case, we're not going to do depth-in-method-call as you mentioned above ☝️ , so I'll let you decide if you've got the motivation to figure out the CSS for that 😅 No big deal either ways 👍

Yup that is right.

I'm reusing all the CSS and DOM structure from the admin side, so changing the CSS just for this code path would be more trouble that its currently worth. Certainly worth revisiting though.